### ISSUE TITLE
Navigate to file/symbol - slow (determining current block)

### ISSUE THREAD

**Issue by davidroth** [dotnet/roslyn#20327]
**Version Used**: 
VS 2017 (15.2)

**Steps to Reproduce**:

1. Press CTRL+, 
2. Search for some method/symbol/file
3. Press enter

I have not found a way to reproduce this deterministically, but I am getting this dialog several times a day.

**Expected Behavior**:

File is opened immediately without a noticeable delay.

**Actual Behavior**:

Dialog "determining current block" shows up with several seconds delay before the real document opens:

![image](https://user-images.githubusercontent.com/338856/27330863-6a0bcf7a-55bb-11e7-991e-41a9ab27a63b.png)

**Comment by CyrusNajmabadi** [comment]
@dpugh @markw-t this is the issue we reported earlier.  

@Pilchie from talking to David, they don't think they can get a fix in for the near future.  One possibility on our side is to compute and cache the result, but return a dummy result until the value is determined.  Or, we can just lie and say that block-searching is always available.  And then, if the editor asks us for the block always succeed.

What do you think?  

I've also run into this and it's *super* annoying.

**Comment by Pilchie** [comment]
> Or, we can just lie and say that block-searching is always available. And then, if the editor asks us for the block always succeed.

I don't understand - isn't them asking for the block how they determine if we're available?  How would we know when to lie and when to tell the truth?

**Comment by Pilchie** [comment]
@sharwell care to take a quick look while @CyrusNajmabadi is on vacation?

**Comment by MattGertz** [comment]
Mondo escrow triage:  moving out.  Agree this is annoying, but let's go for the real fix.

**Comment by SuEric** [comment]
Happened to me too! When trying to open a file tho. 

Is this being fixed???

**Comment by Sword-Smith** [comment]
I have not experienced this problem in VS17 but when I installed the VS19 preview edition, I encountered this problem. I wonder if the preview edition is unable to handle the new switch syntax introduced in C# 8. This problem usually arises when I edit or write files with this new switch syntax.

**Comment by Spider-IT** [comment]
Visual Studio Community 2022 (64-bit) version 17.6.4, .NET Framework version 4.8.04084, Windows 10.
In a routine, I pasted a list of words (28), one word in a line, then (block edit) put 'String ' in front of and ' = DB.ReadSetting("", con);' after it (the empty strings were to fill in the next step).
VS reacted slower on every character entered and then stopped reacting completely while showing the 'Determining current block' message.
The message was there for over 30 minutes with 50+% CPU load (temporarily up to 96%, later down to about 30%).
When I clicked 'Cancel' it took just one second for the same message to appear again, tried that lots of times, then the message didn't come back again (cancelled about 28 times, as much as the new lines).
I could then type again, but nothing else seemed to work (double click on a word to select it, code hinting, ...).
Saving my work was possible, but ending VS didn't work, I had to kill it through the Task Manager.
After restarting VS and loading the project again, everything seems to work again, but all closed code blocks were open again.
Hope this helps...

**Comment by sharwell** [comment]
@Spider-IT We would need a performance trace captured and submitted following these instructions in order to start investigating this:
https://docs.microsoft.com/en-us/visualstudio/ide/how-to-increase-chances-of-performance-issue-being-fixed?view=vs-2019#slowness-and-high-cpu-issues

**Comment by Spider-IT** [comment]
@sharwell every time I click the 'Report a Problem...', a page opens in the Brave browser, and immediately after that it shows the
![Screenshot](https://github.com/dotnet/roslyn/assets/8896694/5cfeb71a-52fd-469c-aeec-a524171c315d)
error.
I have made a video of the initial error, which doesn't go exactly this far again after the restart of VS (which ran for about a week before the last error), but shows it getting extremely slow on just reproducing the error.

https://github.com/dotnet/roslyn/assets/8896694/92a264c4-412c-4ee2-bf4a-63db11a9cdf6

The starting screen shows what I added before, and then I add the very same just above that.
VS goes so slow, that it can't even correct the syntax highlighting.
